lib/node_modules/@stdlib/string/first/lib/main.js

```diff
@@ -0,0 +1,101 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var isNonNegativeInteger = require( '@stdlib/assert/is-nonnegative-integer' ).isPrimitive;
+var nextGraphemeClusterBreak = require( '@stdlib/string/next-grapheme-cluster-break' );
+var format = require( '@stdlib/string/format' );
+
+
+// MAIN //
+
+/**
+* Returns the first (visual) character(s) of a string.
+*
+* @param {string} str - input string
+* @param {NonNegativeInteger} [n=1] - number of characters to return
+* @throws {TypeError} must provide a string primitive
+* @throws {TypeError} second argument must be a nonnegative integer
+* @returns {string} updated string
+*
+* @example
+* var out = first( 'last man standing' );
+* // returns 'l'
+*
+* @example
+* var out = first( 'presidential election' );
+* // returns 'p'
+*
+* @example
+* var out = first( 'javaScript' );
+* // returns 'j'
+*
+* @example
+* var out = first( 'Hidden Treasures' );
+* // returns 'H'
+*
+* @example
+* var out = first( '🐶🐮🐷🐰🐸', 2 );
+* // returns '🐶🐮'
+*
+* @example
+* var out = first( 'foo bar', 4 );
+* // returns 'foo b'
+*/
+function first( str, n ) {
+	if ( !isString( str ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', str ) );
+	}
+	if ( str.length == 0 ) {
+		return '';
+	}
+	var out;
+	const symbols = [...str]
```

Comment:
This works; however, project convention is to use ES5, which means using the spread operator is not permitted.

Instead, we can use `@stdlib/string/next-grapheme-cluster-break` to get the repeatedly get the indices for visual characters and then, instead of a loop as done below with string concatenation, we can use `str.substring( 0, idx )`, which should be preferable in any event due to less GC pressure arising from temporary strings.